recipes/PeoplesSpeech/peoples_speech_prepare.py

```diff
@@ -0,0 +1,398 @@
+    subsets : list
+        Target subset. People's speech contains multiple subsets, which must be
+        loaded invidividually and then concatenated. E.g. 'clean', 'clean_sac',
+        'dirty' or 'dirty_sa'. E.g. to combine  ['clean', 'dirty'].
+    skip_prep : bool, optional
+        If True, the data preparation will be skipped, and the function will return immediately.
+
+    Returns
+    -------
+    None
+    """
+
+    if not os.path.isfile(hf_download_folder):
+        msg = "You must download the dataset with HuggingFace before starting "
+        msg += (
+            "this recipe. Please check the HuggingFace hub of people's speech."
+        )
+        raise ValueError(msg)
+
+    if skip_prep:
+        logger.info("Skipping data preparation as `skip_prep` is set to `True`")
+        return
+
+    if len(subsets) == 0:
+        raise ImportError(
+            "At least one People's speech subset must be specified."
+        )
+
+    # Setting output paths
+    save_output = {}
+    splits = ["train", "validation", "test"]
+    for split in splits:
+        save_output[split] = os.path.join(save_folder, str(split) + ".csv")
+
+    # check if the data is already prepared
+    if skip_csv(save_output):
+        logger.info("Skipping preparation, completed in previous run.")
+        return
+    else:
+        logger.info("Starting data preparation...")
+
+    hf_dataset = load_and_concatenate_datasets(subsets, hf_download_folder)
+
+    logger.info(
+        f"Preparing CSV of the Peoples Speech dataset in {save_folder}..."
+    )
+
+    os.makedirs(save_folder, exist_ok=True)
+
+    i = 0
+    for split, output in save_output.items():
+        logger.info(f"Starting creating {output} using {split} split.")
+        HF_create_csv(output, hf_dataset[i], split)
+        i += 1
+
+    logger.info("Data preparation completed!")
+
+
+def load_and_concatenate_datasets(subsets, hf_download_folder):
+    """Load/download and concatenate all the specified subsets from People's
+    speech
+    """
+
+    try:
+        import datasets
+        from datasets import concatenate_datasets, load_dataset
+    except ImportError:
+        raise ImportError("HuggingFace datasets must be installed.")
+
+    # Managing the download dir as HF can be capricious with this.
+    if "HF_HUB_CACHE" in os.environ:
+        hf_caching_dir = os.environ["HF_HUB_CACHE"]
+    elif "HF_HOME" in os.environ:
+        hf_caching_dir = os.environ["HF_HOME"]
+    else:
+        hf_caching_dir = os.environ["XDG_CACHE_HOME"]
+
+    if hf_caching_dir != hf_download_folder:
+        msg = "HuggingFace HF_HUB_CACHE or HF_HOME is not equal to the given"
+        msg += " hf_download_folder. Make sure to set these variables properly."
+        raise Exception(msg)
+
+    logger.info("Loading dataset from: " + str(hf_caching_dir))
+
+    import multiprocessing
+
+    nproc = (
+        multiprocessing.cpu_count() // 2 + 1
+    )  # we don't want to use all cores
```

Comment:
shouldn't it be something that the user is supposed to set?

Reply:
No, I found a few bugs with HuggingFace multiprocessing. It shouldn't be too high or it starts to go blblblblblbl. Fixing it like that is good imho .